variant/tokenizers/amino_acid_substitution.py

```diff
@@ -0,0 +1,78 @@
+"""A module for Amino Acid Substitution Tokenization."""
+from typing import Optional
+from variant.schemas.token_response_schema import AminoAcidSubstitutionToken,\
+    TokenMatchType
+from .polypeptide_sequence_variant_base import PolypeptideSequenceVariantBase
+
+
+class AminoAcidSubstitution(PolypeptideSequenceVariantBase):
+    """Class for tokenizing Amino Acid Substitution."""
+
+    def match(self, input_string) -> Optional[AminoAcidSubstitutionToken]:
+        """Return a AminoAcidSubstitutionToken match if one exists.
+
+        :param str input_string: The input string to match
+        :return: A AminoAcidSubstitutionToken if a match exists.
+            Otherwise, None.
+        """
+        input_string = str(input_string)
+
+        psub_parts = None
+        self.psub = {
+            'amino_acid': None,
+            'position': None,
+            'new_amino_acid': None
+        }
+
+        if input_string is None:
+            return None
+
+        if '.' in input_string:
+            if not input_string.startswith('p.'):
+                return None
+            p_count = input_string.count('p.')
+            if p_count == 1:
+                psub_parts = self.splitter.split(input_string)
+            elif p_count == 2:
+                psub_parts = input_string.split()
+        else:
+            psub_parts = self.splitter.split(input_string)
+
+        self._get_psub(psub_parts)
+
+        if all(self.psub.values()):
+            amino_acids = {self.psub['amino_acid'],
+                           self.psub['new_amino_acid']}
+
+            if not self._is_valid_amino_acid(amino_acids):
+                return None
+
+            return AminoAcidSubstitutionToken(
+                token=input_string,
+                input_string=input_string,
+                match_type=TokenMatchType.UNSPECIFIED.value,
+                ref_protein=self.psub['amino_acid'],
+                alt_protein=self.psub['new_amino_acid'],
+                position=self.psub['position']
+            )
+
+        return None
+
+    def _get_psub(self, psub_parts):
+        """Return Protein Substitution tokens.
+
+        :param list psub_parts: The split input string
+        """
```

Comment:
Docstring is a little misleading since the method doesn't really return anything IMO

Reply:
Forgot to fix it after changing what the method did! Updating now